health.go

```diff
@@ -229,3 +244,22 @@ func (h *health) updateResult(
 	h.results[name] = result
 	return result
 }
+
+func (h *health) registerShutdownHook(cancel context.CancelFunc) {
```

Comment:
I'm not sure this is a good idea. `main` should call `Stop` or `Close` on the health check instead.

Update: rephrasing: this is a bad idea. 🙂

Reply:
I agree with this. @dmarkhas we can add a stop() or close() functionality to the health object in another PR, and have the users decide what to do with it

Reply:
Ok, let's go with context.TODO for now.